### ISSUE TITLE
clicking link to community shows 0 posts and not logged in

### ISSUE THREAD

**Issue by Die4Ever** [LemmyNet/lemmy-ui#2347]
### Requirements

- [X] This is a bug report, and if not, please post to https://lemmy.ml/c/lemmy_support instead.
- [X] Please [check](https://github.com/LemmyNet/lemmy-ui/issues) to see if this issue already exists.
- [X] It's a single bug. Do not report multiple bugs in one issue.
- [X] It's a frontend issue, not a backend issue; Otherwise please create an issue on the [backend repo](https://github.com/LemmyNet/lemmy) instead.

### Summary

When I click the link, it opens with me not logged in and showing 0 posts. If I open https://programming.dev and browse to the community manually then it correctly shows posts and I am still logged in.

### Steps to Reproduce

1. click https://programming.dev/c/gaming@lemmy.zip

### Technical Details

when clicking the link: ![image](https://github.com/LemmyNet/lemmy-ui/assets/30947252/96a23dc6-1f60-4be8-ad3c-a01922ce2798)

when browsing to the community manually: ![image](https://github.com/LemmyNet/lemmy-ui/assets/30947252/6586af72-524e-43e1-821d-fe79fae55cca)

this might only happen for logged in users, I can't reproduce with other links like https://lemmy.ml/c/gaming@lemmy.zip but if you're logged in there then you might get the same issue?

### Lemmy Instance Version

0.19.1

### Lemmy Instance URL

https://programming.dev

**Comment by Die4Ever** [comment]
Still happening on 0.19.3

**Comment by Die4Ever** [comment]
Maybe related to other issues with initial page state like #1953 and #1375

**Comment by Die4Ever** [comment]
pages loading thinking you're not logged in can be a bit disorienting

I clicked a link from https://lemmyverse.net/communities, it opened in my home instance, but the page loaded without considering me logged in for some reason

![image](https://github.com/LemmyNet/lemmy-ui/assets/30947252/b5294032-8572-4354-aef6-ce20ad462bbe)

refreshing the page fixes it, but I think new users would be really thrown for a loop by this

**Comment by Nutomic** [comment]
Maybe also related to caching https://github.com/LemmyNet/lemmy-ui/pull/3251

**Comment by dessalines** [comment]
Yep its likely either caching or nginx caching related. Lots of instances use a nginx cache for non-logged-in users.
